Fable.Remoting.Client/Proxy.fs

```diff
@@ -148,12 +152,15 @@ module Proxy =
                [ box arg0;box arg1;box arg2;box arg3;box arg4;box arg5;box arg6;box arg7;box arg8;box arg9;box arg10;box arg11;box arg12;box arg13;box arg14;box arg15 ] 
                |> List.take typeCount
             promise {
+                let! token = options.TokenCallback()
+                let authHeader = if isNull token || token = "" then [] else [Authorization token]
+
                 // Send RPC POST request to the server
                 let requestProps = [
                     Body (unbox (toJson data))
                     Method HttpMethod.POST
                     Credentials RequestCredentials.Sameorigin
-                    requestHeaders (customHeaders@options.Headers)
+                    requestHeaders (customHeaders@options.Headers@authHeader)
```

Comment:
Wouldn't that possibly override a previously defined auth header?

Reply:
Wouldn't that end in two auth headers since `options.Headers` is a `list` and not a `set` ? Not sure if that's better though ;) 
It wasn't handled before either but I could check if an auth header is already provided and if so ignore the token.

Reply:
Probably replacing it would be better, but I think we should make the authorization a part of the option again. Should be easier to handle.

Reply:
So adding it to the `Headers` when `with_token` is called as before  + adding or replacing an existing token from the `Headers`  when `callback` returns `Some` ? Maybe we should make the callback an option then too? In order to not await a promise when `with_token_callback` isn't called.

Reply:
I did some changes for it to be an option too, also some changes to get the token once, every time (the default) or only on error.